### PR TITLE
Cleaned up/Fixed some bugs in the remote inspector code.

### LINKED ISSUE #22377: "No loader found for resource: res://" when script defines a class
**Godot version:**

8b2eeaa724829445b1f8b65855a816003df4f2ba

**OS/device including version:**

Windows 7 SP1

**Issue description:**

Message "No loader found for resource: res://" gets spammed to Output when selecting a node with a script that defines a class.

**Steps to reproduce:**
Run reproduction project and select 'Node' in Scene window.

**Minimal reproduction project:**

https://github.com/hilfazer/Projects/tree/EngineIssues/Godot/NoLoaderResourceClass

### PR DESCRIPTION
**Pull request by Piet-G**
Cleaned up/Fixed some bugs in the remote inspector code.

This makes some of my previous code cleaner while resolving a bunch of bugs, this also removed an old ugly hack that was already there.

Fixes #22377

### CODE REVIEW

File: editor/script_editor_debugger.cpp [hunk 1]
```diff
@@ -493,23 +493,31 @@ void ScriptEditorDebugger::_parse_message(const String &p_msg, const Array &p_da
...
+			Variant::Type TYPE = var.get_type();
+
+			if (pinfo.type == Variant::OBJECT) {
+				if (var.is_zero()) {
+					var = RES();
+				} else if (var.get_type() == Variant::STRING) {
+					var = ResourceLoader::load(var);
+
+					if (pinfo.hint_string == "Script")
+						debugObj->set_script(var);
```
Comment (DualMatrix) [inline editor/script_editor_debugger.cpp]: This is the only line I'm really unsure about, this was the only way I could make scripts show up in the in the inspector but it doesn't look quite right. What do you guys think?

File: editor/script_editor_debugger.cpp [hunk 2]
```diff
@@ -493,23 +493,31 @@ void ScriptEditorDebugger::_parse_message(const String &p_msg, const Array &p_da
 			pinfo.usage = PropertyUsageFlags(int(prop[4]));
 			Variant var = prop[5];
 
+			Variant::Type TYPE = var.get_type();
```
Comment (akien-mga) [inline editor/script_editor_debugger.cpp]: Seems unused.

Comment (DualMatrix) [reply editor/script_editor_debugger.cpp]: Yup, that was a remnant from debugging. Removed it.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**